gazebo_plugins/include/gazebo_plugins/gazebo_ros_template.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_PLUGINS__GAZEBO_ROS_TEMPLATE_HPP_
+#define GAZEBO_PLUGINS__GAZEBO_ROS_TEMPLATE_HPP_
+
+#include <gazebo/common/Plugin.hh>
+
+// For std::unique_ptr, could be removed
+#include <memory>
+
+namespace gazebo
+{
+// Forward declaration of private data class.
+class GazeboRosTemplatePrivate;
+
+class GazeboRosTemplate : public ModelPlugin
```

Comment:
Would be helpful to mention that other base classes (WorldPlugin, SensorPlugin, etc) are possible

Reply:
Good idea, added mentions: https://github.com/ros-simulation/gazebo_ros_pkgs/pull/783/commits/2bc125b356bcaa39aef6c139c48e697f1f701c5e